### DISCUSSION TITLE
What are the Modern Search PnP Search parameters?

### DISCUSSION THREAD

**Discussion by RuthChurchman** [microsoft-search/pnp-modern-search#2637] [Q&A]
We have External Developers that have provided us with custom Search web parts, based on the Modern Search PnP web parts which are in test for rolling out to our intranet, but we are getting strange results when using them for a People Search. e.g If searching for the name "Nancy", if we put in a partial name "Nan" (searching without speech marks) we are getting "Nancy" but are also getting other results that do not contain "nan" consecutively in any of the fields on SharePoint user profiles. The Search Results are set to bring back search results from all People fields using "*". The OOTB search does not give this issue. What are the parameters of the Modern Search PnP web parts that could be causing this issue?  e.g. is it doing a "fuzzy" search for something similar to "nan", or those letters in a different order, or what? Just trying to reduce the extraneous "noise" in the search results for the user.

**Comment by gretchunkim** [comment]
If you take a look at the PnP Search Results webpart, it provides some clues to what you're seeking for- it's listed under "Selected properties. All these are parameters. In addition, you can narrow it down based on Query template section.  

Here's even better documentation to answer your search parameter question: https://microsoft-search.github.io/pnp-modern-search/usage/search-results/data-sources/ 

![image](https://user-images.githubusercontent.com/8564035/206786033-aaa2acae-a60f-4763-b3fb-f58619fb7b4a.png)

**Reply by RuthChurchman** [reply]
@gretchunkim Thanks, I have looked at these, see my reply to wobba below for our current settings

**Comment by wobba** [comment]
@RuthChurchman OOB people search in Microsoft Search uses a different way to search for people, not only profiled for UPA. Thus results will not be 100% the same.

Can you also share the query template and setup used for people search?

Also curious why dev's have given you a modified version of the web parts and not using them as is? Because any change makes it harder if you want to upgrade later.

**Reply by RuthChurchman** [reply]
@wobba The web parts have been customised because we use a 3 letter code in the Alias/Initials field interchangeably with our Names, so we needed to surface that info higher in the user profile, so it now displays in brackets next to our names, and we also surfaced a few other fields to be seen immediately in the People Results, and some other fields deeper in the info card structure. This is due to upgrading from a customised People Search on SP2010, and not wanting to lose some of the original features. 
![Search Results Screenshot](https://user-images.githubusercontent.com/83355220/207046727-d1532840-ea9d-4b4c-aa43-ae51d3cafffe.jpg)

The settings are as follows:
Query template
{searchTerms}*

Result Source Id / Scope|Name: 
fcb3bc5d-76f8-4fb6-9ca2-e42ea5558522

Selected properties: 
AuthorOWSUSER
Contentclass
ContentTypeID
Created
CreatedBy
DefaultEncodingURL
DocumentLink
FileType
HitHighlightedSummary
JobTitle
NormListID
NormSiteID
NormUniqueID
NormWebID
OWStaximetadatalltagsinfo
Path
SiteTitle
SPSiteURL
Title
UniqueID
UserName
WorkPhone

Edit layout slots for this data source:
Slot Name: UserDisplayName = Slot Field: Title
Slot Name: PersonQuery = Slot Field: UserName

![image](https://user-images.githubusercontent.com/83355220/207047268-20761c86-c25c-435c-ae7b-a5ec6c352c4b.png)

![image](https://user-images.githubusercontent.com/83355220/207047325-c830de94-1033-4b8c-986c-c1fc097bf7cc.png)

![image](https://user-images.githubusercontent.com/83355220/207047382-97bf53bd-dd50-47c9-a88d-0831da03ab8c.png)

![image](https://user-images.githubusercontent.com/83355220/207047537-bbc9af92-3380-4d92-ae38-98687c8522c5.png)

![image](https://user-images.githubusercontent.com/83355220/207047609-ccf2f659-fd93-497b-8e17-349ac8f5b815.png)

 
![image](https://user-images.githubusercontent.com/83355220/207047686-784ba7e4-cdaf-4b86-9d82-eef0ea3aee8b.png)

**Reply by wobba** [reply]
So the customization is in the display template, not the web part then - which is good, and the right approach :) Then upgrades are not an issue.

You are better off changing your query template to: `({searchTerms}*) OR ({searchTerms})`.

And if you get results on `nan*` that means there is data in some profile property matching that term. Should be impossible to get anything where that is not true. If you only want wildcards to match names you can change the query template to:

`Title:({searchTerms}*) OR ({searchTerms})`

**Reply by RuthChurchman** [reply]
I tried changing the query template to `({searchTerms}*) OR ({searchTerms})`, but it just changed to `{searchTerms}*` and I still got the same results. 

I have checked the SPO user profiles from the `nan*` search results on the ones where it wasn't apparent where `nan*` was in their profiles, but did not find `nan` in any field, even ending one word and starting another with those letters.

We don't want to just search the title field we want to be able to combine field searches, e.g. `Neil HR` which works OK.

If we can't resolve the extra search results we need to be able to explain why it is happening to the users and suggest another way to search, which is why I was asking about the search parameters to understand why we are getting those results.

**Reply by gretchunkim** [reply]
@RuthChurchman For parameters, if I were you start narrowing down selected properties- these are considered parameter. Eliminate one by one (meaning deselect them so you don't including as your search parameter) and see if your search result as your troubleshooting step if you haven't done already.  wobba provided root cause answer to your question in one line (go wobba!).  

![image](https://user-images.githubusercontent.com/8564035/208724365-b66028e1-ab8d-4127-8d41-2da1a3d6708b.png)

**Comment by gretchunkim** [comment]
@RuthChurchman Click "debug" from layout option.  Then try to see if you get any field that contains Na or close to that and filter it out if you can. I do see on my end that GeoLocationSource  item contains "NAM".

**Reply by RuthChurchman** [reply]
Thanks @gretchunkim I hadn't tried the debug layout option. I was then able to search for "nan" in the code. It was found in the first 3 results, as expected, but not anywhere in the following 5 results, so still doesn't explain why they are there. "nan" does not appear in the "GeoLocationSource" for our profiles

**Reply by gretchunkim** [reply]
@RuthChurchman  "nan" exactly match won't be but it will show up in people search debug in SharePoint as datasource but it will be a part of "nam". I looked into what this is referring to, and it looks like a part of string reference to Delve.  

As for your query template field, if you want to target UserName field (or any other fields), use the following: 

`UserName:({searchTerms}*)`

If you want to combine search based on job titles or user's name then this:
`UserName:({searchTerms}*) OR JobTitle:({searchTerms}*) `

The idea is to focus on the field where you want to search to look information into (query).  

Additionally, look into each schema property and which field is selected query. You'll want to deselect them once you identified why certain results returning more than expected.
